app/workers/generic_worker.rb

```diff
@@ -0,0 +1,14 @@
+class GenericWorker
+
+  include Sidekiq::Worker
+
+  def perform(klass_name, *args)
+    klass = klass_name.constantize
+    options = args.extract_options!
```

Comment:
Suggestion for clarity: The signature could be explicitly `klass_name, klass_method, method_args`.